packages/backend/src/open_payments/receiver/model.test.ts

```diff
@@ -0,0 +1,137 @@
+import { IocContract } from '@adonisjs/fold'
+import { Knex } from 'knex'
+
+import { createTestApp, TestContainer } from '../../tests/app'
+import { Config } from '../../config/app'
+import { initIocContainer } from '../..'
+import { AppServices } from '../../app'
+import { createIncomingPayment } from '../../tests/incomingPayment'
+import { createPaymentPointer } from '../../tests/paymentPointer'
+import { truncateTables } from '../../tests/tableManager'
+import { ConnectionService } from '../connection/service'
+import { Receiver } from './model'
+import { IncomingPaymentState } from '../payment/incoming/model'
+
+describe('Receiver Model', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let knex: Knex
+  let connectionService: ConnectionService
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+    connectionService = await deps.use('connectionService')
+    knex = await deps.use('knex')
+  })
+
+  afterEach(async (): Promise<void> => {
+    jest.useRealTimers()
+    await truncateTables(knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe('fromIncomingPayment', () => {
+    test('creates receiver', async () => {
+      const paymentPointer = await createPaymentPointer(deps)
+      const incomingPayment = await createIncomingPayment(deps, {
+        paymentPointerId: paymentPointer.id
+      })
+
+      const receiver = Receiver.fromIncomingPayment(
+        incomingPayment.toOpenPaymentsType({
+          ilpStreamConnection: connectionService.get(incomingPayment)
+        })
+      )
+
+      expect(receiver.asset).toStrictEqual({
+        code: incomingPayment.asset.code,
+        scale: incomingPayment.asset.scale
+      })
+      expect(receiver.incomingAmount).toBeUndefined()
+      expect(receiver.receivedAmount).toStrictEqual({
+        assetCode: incomingPayment.asset.code,
+        assetScale: incomingPayment.asset.scale,
+        value: BigInt(0)
+      })
+      expect(receiver.ilpAddress).toEqual(expect.any(String))
+      expect(receiver.sharedSecret).toEqual(expect.any(Buffer))
+      expect(receiver.expiresAt).toEqual(incomingPayment.expiresAt)
```

Comment:
I had that originally, but because it wasn't a normal `Object` but a `Receiver` class, my assertion was failing. Going to try again, maybe I was just using a wrong jest method

Reply:
Ended up with:
```ts
      expect(receiver).toEqual({
        assetCode: incomingPayment.asset.code,
        assetScale: incomingPayment.asset.scale,
        incomingAmountValue: undefined,
        receivedAmountValue: BigInt(0),
        ilpAddress: expect.any(String),
        sharedSecret: expect.any(Buffer),
        expiresAt: incomingPayment.expiresAt
      })
```

trick was not using the `getters` like `receiver. asset`